crates/ty_project/src/lib.rs

```diff
@@ -574,30 +583,32 @@ impl<'a> IntoIterator for &'a ProjectFiles<'a> {
 
     fn into_iter(self) -> Self::IntoIter {
         match self {
-            ProjectFiles::OpenFiles(files) => ProjectFilesIter::OpenFiles(files.iter()),
-            ProjectFiles::Indexed(indexed) => ProjectFilesIter::Indexed {
-                files: indexed.into_iter(),
-            },
+            ProjectFiles::OpenFiles(files) => {
+                ProjectFilesIter::OpenFiles(files.as_ref().map(|files| files.iter()))
```

Comment:
The `HashSet` iterator implements default. This allows us to simplify the type in `OpenFiles` to just `hash_set::Iter`.


```suggestion
                ProjectFilesIter::OpenFiles(files.map(|files| files.iter()).unwrap_or_default())
```